strings/longestvalidparanthesis.py

```diff
@@ -0,0 +1,18 @@
+def longestValidParentheses(self, s: str) -> int:
```

Comment:
- change function name from `longestValidParentheses' to 'longest_valid_parentheses'
- write test cases. checkout other files to get a hint on testcases
- explain the working of function in comments